server/context.go

```diff
@@ -131,6 +134,37 @@ type Context struct {
 	TimeUntilStoreDead time.Duration
 }
 
+func getDefaultCacheSize() int64 {
+	mem := sigar.Mem{}
+	err := mem.Get()
+	if err != nil {
+		log.Infof("system available memory info is invalid, setting rocksdb cache size to 512MB %s", err)
+		return 512 << 20
+	}
+	sysAvlMem := int64(mem.Total)
+	if sysAvlMem < 0 {
```

Comment:
@tschottdorf Ideally it should never happen. But since we are converting from uint64 to int64 there is a chance of overflow. So as a precautionary measure I wrote this.